server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureState.java

```diff
@@ -54,4 +74,31 @@ public long getLimitReachedCount() {
     long incrementLimitReachedCount() {
         return limitReachedCount.incrementAndGet();
     }
+
+    public AtomicReference<TokenBucket> getRateLimiter() {
+        return rateLimiter;
+    }
+
+    public AtomicReference<TokenBucket> getRatioLimiter() {
+        return ratioLimiter;
+    }
```

Comment:
Instead of exposing the atomic reference, can you simply return the value held by that reference? It's much safer as it avoids unintentional modifications.

```
public TokenBucket getRateLimiter() {
    return rateLimiter.get();
}
```

Reply:
Makes sense. Refactored.